language/move-analyzer/src/symbols.rs

```diff
@@ -214,6 +218,8 @@ pub struct Symbols {
     file_use_defs: BTreeMap<PathBuf, UseDefMap>,
     /// A mapping from file hashes to file names
     file_name_mapping: BTreeMap<FileHash, Symbol>,
+    /// A mapping from filePath to function definitions
+    pub file_functions: BTreeMap<PathBuf, FunctionUseDefMap>,
```

Comment:
Instead of making it public, perhaps you'd consider implementing the right accessor functions for the `Symbols` type? After all you already pass a read-only reference to `Symbols` to `on_completion_request`.